shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -286,21 +286,31 @@ - (NSArray*)accessibilityChildren {
 @implementation FlutterViewController {
   // The project to run in this controller's engine.
   FlutterDartProject* _project;
+
+  std::shared_ptr<flutter::AccessibilityBridgeMac> _bridge;
 }
 
 @dynamic view;
+@dynamic accessibilityBridge;
 
 /**
  * Performs initialization that's common between the different init paths.
  */
-static void CommonInit(FlutterViewController* controller) {
-  if (!controller->_engine) {
-    controller->_engine = [[FlutterEngine alloc] initWithName:@"io.flutter"
-                                                      project:controller->_project
-                                       allowHeadlessExecution:NO];
+static void CommonInit(FlutterViewController* controller, FlutterEngine* engine) {
+  if (!engine) {
+    engine = [[FlutterEngine alloc] initWithName:@"io.flutter"
```

Comment:
Could you elaborate as to why we are allocating a `FlutterEngine` if the client of this call does not pass a `FlutterEngine?`

Reply:
This is a convention with a long history. In this way, the user can simply create a `FlutterViewController` and assigns it to the window, and Flutter just works. The user doesn't have to know that `FVC` creates an engine, and the engine runs a Dart VM, etc. In fact, this is the sample runner that Flutter creates for a default user.

(Although IIRC, the history worked the other way around. At first Flutter only had a `FlutterViewController`, and `FlutterEngine` was split from it afterwards.)

Reply:
And also, `CommonInit` is only called by the various initializers of `FVC`. Your question might be better phrased as "why we are allocating a `FlutterEngine` for `FVC`'s initializers that do not receive an engine".

Reply:
Sounds good, I had concerns about multiple engines being created, but I think the API is clear enough that it shouldn't be the case. I would hope that the allocation, if the `FlutterEngine` passed is null, is documented somewhere. That's all.

Reply:
I think it's intentionally designed that multiple `FlutterEngine` instances can exist in a single context (for example, in the case of [`FlutterEngineGroup`](https://api.flutter.dev/javadoc/io/flutter/embedding/engine/FlutterEngineGroup.html), a workaround for multi-view in the single-view era), otherwise we would've used a static variable for the singleton engine. 

The documentation has been added to the initializers.